### ISSUE TITLE
create a dedicated Home page for docs at dvc.org/doc (design part left)

### ISSUE THREAD

**Issue by jorgeorpinel** [iterative/dvc.org#1073]
In a reorganization of docs (#1051) we're creating a Home page that /doc redirects to.

We'd like it to BE in /doc. But keeping the **Home** entry in the nav sidebar. Is this easy with the current Gatsby engine?

---------

- [x] It should have dedicated icon. Related https://github.com/iterative/dvc.org/issues/1053.
- [x] Putting it in `dvc.org/doc` most likely means it's a separate page, rather than a regular Markdown file.
- [ ] Design. It should have a table with existing sections, with graphics, etc. Again, probably means it's a separate page rather than a regular one.
- [x] On the sidebar it should stay separate under the search bar.
- [x] Main topics should have a special CSS design (after #1074 is merged):
  ![image](https://user-images.githubusercontent.com/1477535/79699242-da06da00-8253-11ea-9dde-5359bfe5cd7b.png)
  > UPDATE: Probably 4 tiles instead, jump to https://github.com/iterative/dvc.org/issues/1073#issuecomment-630552134

  Good example (at least the middle section) in https://cloud.google.com/sql/docs/mysql:
  ![image](https://user-images.githubusercontent.com/1477535/80674319-2d103680-8a77-11ea-9cfb-9b50c02d82af.png)

**Comment by jorgeorpinel** [comment]
But keeping the **Home** entry in the nav sidebar...

**Comment by iAdramelk** [comment]
@pavelgrinchenko can you check this issue please?

**Comment by pavelgrinchenko** [comment]
@jorgeorpinel In the current codebase you can't do it without changes in the `src/utils/sidebar.js`. If you want you can show your branch with changes and I'll do it for you.

**Comment by pavelgrinchenko** [comment]
I prepared patch. You can get it [here](https://github.com/iterative/dvc.org/commit/69ec1e82), and then just set something like this in sidebar's config
```
{
    "label": "Home",
    "slug": "",
    "source": "home/index.md"
}
```

**Comment by jorgeorpinel** [comment]
Thanks @pavelgrinchenko that works! Cherry-picked your patch into my working branch now :)
See https://github.com/iterative/dvc.org/compare/9ddaf88..ef4ece6

**Comment by shcheklein** [comment]
Updated the title and description. I think Home page is a separate sub-project and a special page. For the SEO purpose we should do it quick

**Comment by jorgeorpinel** [comment]
>  On the sidebar it should stay separate under the search bar.

So first step would be moving the current /doc/home to /doc I think, but not understanding what you mean to do with the Home entry in the sidebar.

**Comment by shcheklein** [comment]
> So first step would be moving the current /doc/home to /doc I think, but not understanding what you mean to do with the Home entry in the sidebar.

Nothing that special. Just make it a bit separate from the main list of regular docs. But I'm not sure, would need to play with it a bit to say how it should look like :)

**Comment by rogermparent** [comment]
Sorry for throwing out a basic fix to the title without looking deep at the content.

My PR moves the current `docs/home` page to `docs`, but there are a few choices we have to make for a page redesign.

First off: Do we make the homepage with Markdown, or `src/pages`? 

Neither are difficult for me to implement, it's more a decision on whether we want to write the content in React or Markdown. We can just use a unique component for home, or I could add the ability for docs pages to use optional template overrides defined in the frontmatter.  
These solutions are ones I'm very experienced with, so I suggest we pick solely on preference and not worry about the implementation dev time- it's very likely a matter of a few hours for me either way. 

I would suggest going with whatever our designer (or whoever wears that hat) is most comfortable with writing content in. We can use an arbitrary consumer component with Markdown, so the technical capabilities of both are essentially equal.

**Comment by jorgeorpinel** [comment]
> Do we want to make the docs home with Markdown, or a src/pages react component?

What do you recommend? I think Ivan's impression is we'll need a special page. I think it would be ideal if it can be kept as md but that may be tricky diven the special design we want for it, which is currently hardcoded horribly (mea culpa) with HTML in the md.

> it's more a decision on whether we want to write the content in React or Markdown
> ability for docs pages to use optional template overrides defined in the frontmatter

So can it be a React page that "loads" an md file for most of it's content, so the content can be easily updated by anyone?

> We can just use a unique component for home

Also this sounds attractive. I'd go with whatever you recommend tbh.

> going with whatever our designer is most comfortable with

Well, I don't think we have a designer per se. For starters we're just talking about those 3 boxes in the middle of the page so hopefully one of us can handle making them look and feel pro.

**Comment by shcheklein** [comment]
@rogermparent I like this simple style for the home page - https://cube.dev/docs , and I can connect us with the guy who has done the design for it.

Re the implementation - up to you. It feels that md can become too limited at some point?

**Comment by rogermparent** [comment]
My personal recommendation would be introducing simple frontmatter-defined templates, keeping md for the content and using a unique home template to get any out-of-markdown design.

As for the question of Markdown becoming limited, we could easily migrate from remark to MDX- it's backward-compatible with Remark and lets us make define our own arbitrary React components that can be used in our Markdown files natively without a bunch of Remark/Rehype trickery.

Re-creating something like that cube.dev page wouldn't be too hard even with Remark. I can put together the "backend" template logic and do the boxes to the best of my ability, then we can take it from there.

**Comment by shcheklein** [comment]
> As for the question of Markdown becoming limited, we could easily migrate from remark to MDX

there were some problems with MDX (built time support, custom plugins, etc). Also, I would say one page does not justify opening the box (there will be no return).

> My personal recommendation would be introducing simple frontmatter-defined templates, keeping md for the content and using a unique home template to get any out-of-markdown design.

similar to what we do for the community page?

> Re-creating something like that cube.dev page wouldn't be too hard even with Remark. I can put together the "backend" template logic and do the boxes to the best of my limited design ability, then we can take it from there.

sounds like a good starting point, we can then ask designer to help us.

**Comment by rogermparent** [comment]
I have a branch tackling this issue at #1268, but I'm starting to think that, given the time-frame for DVC 1.0, we should take a two-phase approach to this issue.

For **phase one**, we get the design and content we want and post it up in whatever is the most doable: I think the best fit is using my template branch to change the docs home's component to a custom component that has full freedom to match any design we want. This solution lets up quickly put up a nice-looking docs home with whatever design we want (including unique queries), while letting us keep the existing metadata that comes from the Markdown file.

Once the page is done, **phase two** is deciding how we want to write blog pages going in the future. The options I see are:

1. Keep the component template solution, treating the docs home as a special case where you edit the template component instead.

2. Port the Component-based homepage design's content data into frontmatter fields (like box definitions) so future edits to the docs home can be done in its markdown page.

3. Formalize custom components. I think going this route necessitates MDX, as the `rehype-react` method has a lot of unfortunate quirks that I didn't remember that makes it pretty distasteful for production.

From what I can see, our current next step for this issue is to put together a design that we want. I know there's been a few examples posted, but I mean something with our content typeset in it as well as any images/icons we'd want to use. 

I can take a design to a component, and we'll have plenty of time to handle phase two after we get the page up.

**Comment by jorgeorpinel** [comment]
> I have a branch tackling this issue at #1268

I've checked that PR's preview and checked most of the check boxes in the description, except that of design, which is what you're suggesting to complete as phase one. That would close out this issue, I think.

> phase two is deciding how we want to write blog pages going in the future

I'm not sure how that relates to the docs home page.

**Comment by rogermparent** [comment]
> I've checked that PR's preview and checked most of the check boxes in the description, except that of design, which is what you're suggesting to complete as phase one. That would close out this issue, I think.

Sure! I meant it as a simple example but if we like it I can take it forward. There's some qualms about the example's custom components implementation, so I intend to re-create whatever design we get in a simple React template component that has full freedom to implement any design.

> > phase two is deciding how we want to write blog pages going in the future
> 
> I'm not sure how that relates to the docs home page.

My bad, I meant to say "docs pages"- special ones like this docs home. Basically, we have to decide how to want to write and handle special docs pages like home from here on out. 

My current plan, at the end of phase one, will make it such that the docs home is a React component in `src/templates` which we edit to change the docs home, and the Markdown file serves as a sort of shim because our current metadata and page generation logic depends on it.  
While this process is a bit inelegant and clunky, it makes up for that in flexibility, speed of implementation, and the possibility of porting the special page content to something more friendly like custom components or frontmatter-based content down the line.

With my plan, we get our new docs home ASAP and then improve the editing experience afterward. They both need to happen, but getting the new page out to users is even more important in my opinion.

In the meantime, I'm still working on adding an alternate bullet icon to home. I'll likely end up doing a similar plan with that, but I don't need to wait for a design for a scope that narrow.

**Comment by jorgeorpinel** [comment]
Gotcha Roger.

> Basically, we have to decide how to want to write and handle special docs pages like home from here on out.

Seems like a nice-to-have right now. I don't know of any other special case other than the "cards" in he home page for now. But maybe you guys have discussed others? Cc @shcheklein 

> make it such that the docs home is a React component in src/templates which we edit to change the docs home, and the Markdown file serves as a sort of shim

I know you guys have discussed this so I won't disagree but have we considered just making these "cards" a special react component that activates autodetecting some basic HTML similar to how tooltips activate with `<abbr>` or expandable sections with `<details>`? Maybe `<div class="card">` or even an XHTML tag such as `<card>`.

**Comment by shcheklein** [comment]
> With my plan, we get our new docs home ASAP and then improve the editing experience afterward. They both need to happen, but getting the new page out to users is even more important in my opinion.

@rogermparent I'm totally fine with that. Up to you. As I mentioned in your PR our goal is a simple page like this for now: https://cube.dev/docs (again, I'm fine if you go with a template for that).

@jorgeorpinel do we have an idea for 4 tiles already? So that I can ask designer to get us some icons. (and probably the home page icon as well)

**Comment by rogermparent** [comment]
> @jorgeorpinel do we have an idea for 4 tiles already? So that I can ask designer to get us some icons. (and probably the home page icon as well)

I second this heartily, 4 tiles is much easier than 3 to make look nice.

**Comment by jorgeorpinel** [comment]
Sure. An easy way is to split the last tile in 2: "**Study** the detailed [inner-workings] of DVC." + "**Understand** each one of its [commands] and [Python functions]."

Or a 4th one could be regarding tutorials, but we're not sure which ones will be staying after the 1.0 release so I can't come up with great text for that right now. Something like "Learn how to used DVC step-by-step with our didactic [tutorials]" ("how to" could later be linked to the How-to section, once created).

**Comment by rogermparent** [comment]
We now have the special bullets in my branch at #1322 (along with circle bullets for leaves), so all that's left now is to make and then implement the design of the page itself.

**Comment by rogermparent** [comment]
If the design doesn't stray too far from what I have in my branch, it would be fastest to just adapt what I have to the design for phase 1 and improve the implementation after. 

If we would be happy with just some slight changes to it I can easily implement them, especially where no other page uses the card components currently.

Replacing the whole page with a component is still doable if we get a design that isn't easily replicated with these MD custom components, but the containers used in MD are slightly embedded into the MD components so untying those removes some of the speed and cleanliness of the component method.

**Comment by jorgeorpinel** [comment]
Lets just apply the 4 tiles/cards suggested in https://github.com/iterative/dvc.org/issues/1073#issuecomment-630692444 for now, maybe? And style it on the go via PR review. That would probably close the remainder of this epic 🙂

**Comment by rogermparent** [comment]
Ah, whoops! [This](https://github.com/iterative/dvc.org/pull/1268) is the PR I was talking about earlier as an example- it has a 4 tiles design and is what I think to be pretty close to what we want. Deploy's up and everything!

**Comment by shcheklein** [comment]
Design is the last pending part of it. And we can modify content a little bit when new get started is ready /cc @jorgeorpinel

**Comment by shcheklein** [comment]
Reopening since it's not done yet.

**Comment by shcheklein** [comment]
Closing this for now. Probably we'll have the full redesign soon.
